### PR TITLE
silenced unused variables

### PR DESCRIPTION
**Pull request by fametrano**
see: https://github.com/bitcoin-core/HWI/pull/367

### CODE REVIEW

File: ckcc/client.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def send_recv(self, msg, expect_errors=False, verbose=0, timeout=3000, encrypt=T
                 print("Rx [%2d]: %r" % (len(resp), b2a_hex(bytes(resp))))
 
             return CCProtocolUnpacker.decode(resp)
-        except CCProtoError as e:
```
Comment (peter-conalgo) [inline ckcc/client.py:164]: this change is ok

File: ckcc/client.py [hunk 2]
```diff
@@ -260,7 +260,7 @@ def mitm_verify(self, sig, expected_xpub):
 
         # If Pycoin is not available, do it using ecdsa
         from ecdsa import BadSignatureError, SECP256k1, VerifyingKey
-        pubkey, chaincode = decode_xpub(expected_xpub)
```
Comment (peter-conalgo) [inline ckcc/client.py:263]: this does not improve code quality

File: ckcc/utils.py [hunk 3]
```diff
@@ -24,14 +24,14 @@ def consume(xfd, tname, fmt, names):
 
     assert dfu_prefix.signature == b'DfuSe', "Not a DFU file (bad magic)"
 
-    for idx in range(dfu_prefix.targets):
+    for _ in range(dfu_prefix.targets):
```
Comment (peter-conalgo) [inline ckcc/utils.py:27]: this does not improve code quality

File: ckcc/utils.py [hunk 4]
```diff
@@ -24,14 +24,14 @@ def consume(xfd, tname, fmt, names):
...
 
         prefix = consume(fd, 'Target', '<6sBI255s2I', 
                                    'signature altsetting named name size elements')
 
         #print("target%d: %r" % (idx, prefix))
 
-        for ei in range(prefix.elements):
```
Comment (peter-conalgo) [inline ckcc/utils.py:34]: this does not improve code quality

**Comment by fametrano** [comment]
Python linters recognize the underscore as a purposefully unused variable, else they raise a warning if the variable is not used later on in the code. As such, I consider the proposed changes to improve code quality.

Of course "your mileage can vary", so if the PR is not well received as it is, I can revert all changes but the approved one.

**Comment by peter-conalgo** [comment]
I'm not against use of underscore as a placeholder, it's just that if a name has been granted to a variable (as in these cases) then the programmer has given it meaning even if it goes unused elsewhere. Removing the name is a reduction in the information contained in the code, and thus not an improvement.

**Comment by fametrano** [comment]
in modern python that information should be available to programmers via type hints and annotations.
As alternative, a comment line could "name" the unused variable in the line below.

Instead, naming a variable sets the expectation of that variable being used later on and this is misleading

**Comment by peter-conalgo** [comment]
1. True
2. Very debatable

Anyway, PR accepted.

**Pull request merged**